File: report/src/main/java/com/vmlens/report/runelementtype/memoryaccesskey/MemoryAccessKey.java
package com.vmlens.report.runelementtype.memoryaccesskey;

import com.vmlens.report.description.DescriptionContext;
import com.vmlens.report.description.NeedsDescriptionCallback;

public interface MemoryAccessKey {

    String asString(DescriptionContext context);
    Long objectHashCode();
    void addToNeedsDescription(NeedsDescriptionCallback callback);

}

File: sync-bug/src/main/scala/com/anarsoft/race/detection/report/element/runelementtype/memoryaccesskey/MemoryAccessKey.scala
package com.anarsoft.race.detection.report.element.runelementtype.memoryaccesskey

import com.anarsoft.race.detection.report.description.{DescriptionContext, NeedsDescriptionCallback}


trait MemoryAccessKey {
  
  def asString(context: DescriptionContext): String
  def objectHashCodeOption : Option[Long]
  def addToNeedsDescription(callback: NeedsDescriptionCallback): Unit
  
}